app/assets/javascripts/components/story/ExpandedStory/index.js

```diff
@@ -29,6 +30,11 @@ export const ExpandedStory = (props) => {
         />
       </div>
 
+      <ExpandedStoryState
+        story={story}
+        onEdit={(newAttributes) => editStory(story.id, newAttributes)}
```

Comment:
defining a function inside a functional component (I mean `() => ()`) means that on every `render` a new function will be created (https://reactjs.org/docs/faq-functions.html#is-it-ok-to-use-arrow-functions-in-render-methods). That implies in a slight performance downfall. In a project like Central, might make a huge difference. 

Hence, I suggest that it's best to stick with the outside definition of functions. Like `const onEditStory = (newAttributes) => editStory(story.id, newAttributes)` and then call 
```
<ExpandedStoryState
         story={story}
         onEdit={onEditStory} />
```

Reply:
Actually, this (premature) optimization won't work. Just saving the function in a `const` won't make it be recreated on every re-render. We can isolate it to a method and bind it (as long as we extract the `story` from the props inside there too), but I don't see a lot of value on doing that right now until we have metrics to show it's necessary. I wouldn't do it right now.

Reply:
What I meant is actually putting `const onEditStory = (newAttributes) => editStory(story.id, newAttributes)` outside  `export const ExpandedStory = (props) => {...}` to prevend recreation on every render. I got the point on optimization but it is also a good approach to prevent repetition like we have in this file in line 35 and 40 `(newAttributes) => editStory(story.id, newAttributes)`